orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/utils/MonikerHelper.java

```diff
@@ -50,4 +52,15 @@ public String getClusterNameFromStage(Stage stage, String fallbackFriggaName) {
     }
     return clusterName;
   }
+
+  static public Moniker monikerFromStage(Stage stage) {
+    Moniker moniker = null;
+    try {
+      moniker = (Moniker) stage.mapTo("/moniker", Moniker.class);
+    } catch (IllegalArgumentException e) {
+      // Moniker doesn't exist
```

Comment:
since context not containng moniker is expected flow, I'd rather see this as `if (stage.context.containsKey("moniker"))` rather than the try-catch